tests/corpus/test_conversation.py

```diff
@@ -27,22 +28,101 @@ def test_instantiate(self, my_convo, convo_utts, convo_meta):
+                                 ([5, 2, 0], 3),
+                                 ([0, 2, 2], 3),
+                                 ([0, 2, None], 3)
+                             ])
+    def test_subconversation_index(self, convo, args, expected_length):
+        index, before, after = args
+        sub = convo._subconversation_by_index(index=index,           # noqa W0212
+                                     before=before,
+                                     after=after)
+        assert isinstance(sub, Conversation)
+        assert len(sub.utterances) == expected_length
+
+    @pytest.mark.parametrize("args, expected_length",
+                             [
+                                 ([0, 0, 0], 2),  # A, B
+                                 ([5, 3000, 3000], 6),  # B,C,E,U,F,H
+                                 ([5, 0, 0], 3),  # C, U, F
+                             ])
+    def test_subconversation(self, convo, args, expected_length):
+        index, before, after = args
+        sub = convo._subconversation_by_time(index=index,           # noqa W0212
+                                     before=before,
+                                     after=after)
+        assert isinstance(sub, Conversation)
+        assert len(sub.utterances) == expected_length
+
+    def test_overlap(self):
+        # entire utterance in window
+        assert Conversation.overlap(80, 120, [90, 110])
+        # beginning of utterance in window
+        assert Conversation.overlap(80, 100, [90, 110])
+        # end of utterance in window
+        assert Conversation.overlap(100, 120, [90, 110])
+        # utterance covers window entirely
+        assert Conversation.overlap(95, 105, [90, 110])
+        assert not Conversation.overlap(
+            120, 140, [90, 110])  # utterance before window
+        assert not Conversation.overlap(
+            70, 80, [90, 110])  # utterance after window
+
+    def test_count_participants(self, convo):
+        assert convo.count_participants() == 4
+        assert convo.count_participants(except_none=True) == 3
+        convo2 = convo._subconversation_by_index(index=0, before=2)  # noqa W0212
+        assert convo2.count_participants() == 2
+        convo3 = convo._subconversation_by_index(index=0)            # noqa W0212
+        assert convo3.count_participants() == 1
+
+    def test_calculate_FTO(self, convo):
+        convo.calculate_FTO()
+        assert convo.metadata["Calculations"]["FTO"] == {
+            "window": 10000, "planning_buffer": 200, "n_participants": 2}
+        convo.calculate_FTO(window=10)
+        assert convo.metadata["Calculations"]["FTO"] == {
+            "window": 10, "planning_buffer": 200, "n_participants": 2}
+        assert convo.utterances[0].FTO is None
+        assert convo.utterances[1].FTO == -100
+        assert convo.utterances[2].FTO is None
+        convo.calculate_FTO(planning_buffer=0)
+        assert convo.utterances[2].FTO == -2499
```

Comment:
Where do -100 and -2499 come from? I would suggest adding a docstring to this test and any other tests that are necessary to explain the setup a bit. Otherwise, it's a bit hard for others to understand what you want to test and why you test it like this.

Reply:
Good point. I've refactored this test and made some more explicit comments.